awx/main/models/unified_jobs.py

```diff
@@ -1384,26 +1407,11 @@ def pre_start(self, **kwargs):
                 self.save(update_fields=['job_explanation'])
                 return (False, None)
 
-        needed = self.get_passwords_needed_to_start()
-        try:
-            start_args = json.loads(decrypt_field(self, 'start_args'))
-        except Exception:
-            start_args = None
-
-        if start_args in (None, ''):
-            start_args = kwargs
+        opts = self.get_start_kwargs()
 
-        opts = dict([(field, start_args.get(field, '')) for field in needed])
-
-        if not all(opts.values()):
-            missing_fields = ', '.join([k for k, v in opts.items() if not v])
-            self.job_explanation = u'Missing needed fields: %s.' % missing_fields
-            self.save(update_fields=['job_explanation'])
+        if opts and (not all(opts.values())):
```

Comment:
we already check for all(opts.values()) in get_start_kwargs
maybe get_start_kwargs should just return None, then we check for `if opts is None` here

Reply:
I was going to say that this would work:

```python
        if opts is None:
            return (False, None)
```

But now I see it won't. The case this intends to distinguish is not the case when there are no `opts`, but the case where we hit something invalid. This isn't really for the `return` but to bypass the `self.job_explanation = ""` later on.

Reply:
ah okay, yeah I see that get_start_kwargs sometimes return None, but that isn't necessarily invalid